### ISSUE TITLE
Profile info ends with error upon submission

### ISSUE THREAD

**Issue by palica** [haxmeister/funtoo-reporter#10]
```
./report send
[Request] ** [http://elk.liguros.net:9200]-[400] [mapper_parsing_exception] object mapping for [profile-info.mix-ins] tried to parse field [null] as object, but found a concrete value, called from sub Search::Elasticsearch::Role::Client::Direct::__ANON__ at ./report line 40. With vars: {'status_code' => 400,'request' => {'ignore' => [],'method' => 'POST','path' => '/funtoo/user/0b3a871c-631f-46b2-a7bc-29111299a9a0%20','mime_type' => 'application/json','serialize' => 'std','qs' => {},'body' => {'profile-info' => {'mix-ins' => ['no-systemd','gnome','hardened'],'arch' => ['x86-64bit'],'flavor' => ['desktop'],'subarch' => ['intel64-ivybridge'],'build' => ['current']},'mem-info' => {'SwapTotal' => 0,'MemTotal' => 3967384,'MemAvailable' => 1286088,'SwapFree' => 0,'MemFree' => 303136},'version-info' => {'ego version' => '2.3.3-r1','python
versions' => ['2.7.14','3.6.4'],'gcc versions' => ['6.4.0'],'portage version' => '2.3.18-r2'},'cpu-info' => {'model name' => 'Intel(R) Core(TM) i3-3217U CPU @ 1.80GHz','siblings' => 4,'cpu MHz' => '1797.68','flags' => ['fpu','vme','de','pse','tsc','msr','pae','mce','cx8','apic','sep','mtrr','pge','mca','cmov','pat','pse36','clflush','dts','acpi','mmx','fxsr','sse','sse2','ss','ht','tm','pbe','syscall','nx','rdtscp','lm','constant_tsc','arch_perfmon','pebs','bts','rep_good','nopl','xtopology','nonstop_tsc','cpuid','aperfmperf','pni','pclmulqdq','dtes64','monitor','ds_cpl','vmx','est','tm2','ssse3','cx16','xtpr','pdcm','pcid','sse4_1','sse4_2','x2apic','popcnt','tsc_deadline_timer','xsave','avx','f16c','lahf_lm','cpuid_fault','epb','pti','tpr_shadow','vnmi','flexpriority','ept','vpid','fsgsbase','smep','erms','xsaveopt','dtherm','arat','pln','pts']},'boot-dir-info' => {'available kernels' => ['bzImage-4.9.8-ck','bzImage-4.9.56-ck','bzImage-4.14.11-ck','bzImage-4.14.14-ck']},'kit-info' => {'gnome-kit' => '3.20-prime','ml-lang-kit' => '1.1-prime','science-kit' => 'master','lang-kit' => '1.1-prime','dev-kit' => '1.1-prime','core-hw-kit' => 'master','lisp-scheme-kit' => '1.1-prime','security-kit' => '1.0-prime','java-kit' => '1.1-prime','python-kit' => '3.6-prime','haskell-kit' => '1.1-prime','perl-kit' => '5.26-prime','text-kit' => 'master','media-kit' => '1.1-prime','xfce-kit' => '4.12-prime','games-kit' => 'master','kde-kit' => '5.10-prime','editors-kit' => 'master','desktop-kit' => '1.1-prime','xorg-kit' => '1.19-prime','core-kit' => '1.0-prime','net-kit' => 'master','nokit' => 'master','ruby-kit' => '1.1-prime','php-kit' => 'master'},'timestamp' => 'Fri Feb  2
09:08:34 2018','kernel-info' => {'ostype' => 'Linux','osrelease' => '4.14.11-ck','version' => '#1 SMP Fri Jan 5 11:02:39 CET 2018'}}},'body' => {'status' => 400,'error' => {'type' => 'mapper_parsing_exception','root_cause' => [{'type' => 'mapper_parsing_exception','reason' => 'object mapping for [profile-info.mix-ins] tried to parse field [null] as object, but found a concrete value'}],'reason' => 'object mapping for [profile-info.mix-ins] tried to parse field [null] as object, but found a concrete value'}}}
```

**Comment by palica** [comment]
[{'type' => 'mapper_parsing_exception','reason' => 'object mapping for [profile-info.mix-ins] tried to parse field [null] as object, but found a concrete value'}],'reason' => 'object mapping for [profile-info.mix-ins] tried to parse field [null] as object, but found a concrete value'}}}

**Comment by haxmeister** [comment]
I was able to get a submission without errors by changing the index it submits too and also by putting the whole thing inside another hash. So this implies that in our tinkering with elastic search we have confusd the system somehow. The error is the return message from elastic search stating that it is trying to process some of the data as an "object" but instead there is a "string". Note that I have changed the output of that function to remove "shortname" and make a straight array instead of an array of key/value pairs, but the system may still be looking for the old structure.
